### PR TITLE
Fix Auto Button in ElvUI

### PR DESCRIPTION
**Pull request by gizzmo**
Uses the ElvUI skin system to make the auto button fit. 

This has been tested in wild, trainer, and pvp battles. And things toggle correctly 

I did have to  have a 1 pixel gap between the two buttons, due to the hover color would not show correctly, due to them being on top of each other.

### CODE REVIEW

**Comment by axc450** [comment]
Ill let @bloerwald take a look at this (as he did some ElvUI stuff prior!)

**Comment by gizzmo** [comment]
So I was looking over the ElvUI API and how they do things, and the standard seem to be to create a method on the Skins module, and then call `S:AddCallbackForAddon('tdBattlePetScript')` to register it. You could also have the function on this module, and register that with `S:AddCallbackForAddon('tdBattlePetScript', Module.ElvUISkin)`

I think doing it this way would allow you to move all the ElvUI specific code into a single function instead of several spots over the file. (IE the `ToolButton`).

**Comment by bloerwald** [comment]
As long as that stupid art frame doesn't reappear and the solution isn't opt-in (i.e. a new skin module rather than just depending on the existing ones), I don't care about how ElvUI handling is done at all, as evidenced by one of my (now removed) comments. I had a not-so-great discussion with ElvUI devs back then so I never cared about how it "should" be done correctly.

If you want to be really sure @axc450, request screenshots of it with 
- ElvUI disabled
- ElvUI enabled, but blizzard frame skinning globally disabled and pet battle skin modules disabled
- ElvUI enabled, but blizzard frame skinning globally enabled but pet battle skin modules disabled
- ElvUI enabled, but blizzard frame skinning globally disabled and pet battle skinning enabled
- ElvUI enabled, and blizzard frame skinning globally enabled and pet battle skinning enabled

I had real "fun" making sure all the combinations work, but it seems that part of the logic is unchanged (the guard-condition in the callback).

**Comment by gizzmo** [comment]
I didnt even think about screenshots. Sorry.

<details>
    <summary>Screen shots </summary>

#### Only PBS
WoWScrnShot_071723_175255

### PBS, ElvUI, with options window visible to show the combinations of `blizzard.enable` and `blizzard.petbattleui`
WoWScrnShot_071723_175843
WoWScrnShot_071723_175810
WoWScrnShot_071723_175729
WoWScrnShot_071723_175638

#### List of addons enabled
WoWScrnShot_071723_175916

#### In PVP pet battle to show the button hides properly
WoWScrnShot_071723_181719
</details>

This is not an opt in thing. If the ElvUI pet skin is enabled then so is this. This just builds on top of what ElvUI is already doing by making PBS fit into the design. During testing I did have the art frame show up again, but noticed that its being made visible with `SetShown` so I blocked that method.

I had a thought after I got your reply. Does this callback really need to be in this file. I think moving it into its own file would be best. Separation of concerns. PetBattle.lua will stay with modifying the default UI, and the new file will contain the ElvUI skin.

File: UI/PetBattle.lua [hunk 1]
```diff
@@ -13,14 +13,48 @@ local Director = ns.Director
...
+
+        ArtFrame2:Hide()
+        ArtFrame2.Show = nop
+        ArtFrame2.SetShown = nop
+
+        local yOffset = E.PixelMode and -1 or 1
+        local gap = E.PixelMode and 1 or 3
+        local xOffset = gap + AutoButton:GetWidth()
+
+        S:HandleButton(AutoButton)
```
[the remark this answers is not in the document]
Comment (gizzmo) [reply UI/PetBattle.lua:41]: no need to do that, ElvUI already does that in their skin.

**Review by bloerwald — APPROVED** [review]
There is the small advantage of knowing you also have to edit the ElvUI hooks if it is in one file, but :shrug:, I don't care about this enough. Fine to merge for me.

**Comment by axc450** [comment]
@gizzmo Thanks so much for taking a look at this. Please add a `CHANGELOG.md` entry under `v1.8` and I will get this merged + released :)

**Comment by gizzmo** [comment]
Added ElvUI as an optional dependency in the toc just to be sure.

**Pull request merged**